File: lazyburn/pricing.py
# Source: https://platform.claude.com/docs/en/about-claude/pricing
# All prices per million tokens.
# cache_5m = 5-minute cache write (1.25x input)
# cache_1h = 1-hour cache write (2x input)
# cache_read = cache hit (0.1x input)

PRICING: dict[str, dict[str, float]] = {
    "claude-sonnet-4-6":         {"input": 3.0,  "cache_5m": 3.75, "cache_1h": 6.0,  "cache_read": 0.30, "output": 15.0},
    "claude-sonnet-4-5":         {"input": 3.0,  "cache_5m": 3.75, "cache_1h": 6.0,  "cache_read": 0.30, "output": 15.0},
    "claude-sonnet-4":           {"input": 3.0,  "cache_5m": 3.75, "cache_1h": 6.0,  "cache_read": 0.30, "output": 15.0},
    "claude-opus-4-7":           {"input": 5.0,  "cache_5m": 6.25, "cache_1h": 10.0, "cache_read": 0.50, "output": 25.0},
    "claude-opus-4-6":           {"input": 5.0,  "cache_5m": 6.25, "cache_1h": 10.0, "cache_read": 0.50, "output": 25.0},
    "claude-opus-4-5":           {"input": 5.0,  "cache_5m": 6.25, "cache_1h": 10.0, "cache_read": 0.50, "output": 25.0},
    "claude-haiku-4-5":          {"input": 1.0,  "cache_5m": 1.25, "cache_1h": 2.0,  "cache_read": 0.10, "output": 5.0},
    "claude-haiku-3-5-20241022": {"input": 0.80, "cache_5m": 1.0,  "cache_1h": 1.6,  "cache_read": 0.08, "output": 4.0},
    "claude-haiku-3-5":          {"input": 0.80, "cache_5m": 1.0,  "cache_1h": 1.6,  "cache_read": 0.08, "output": 4.0},
}

_FALLBACK = {"input": 3.0, "cache_5m": 3.75, "cache_1h": 6.0, "cache_read": 0.30, "output": 15.0}


def get_pricing(model: str) -> dict[str, float]:
    if model in PRICING:
        return PRICING[model]
    for key in PRICING:
        if model.startswith(key) or key.startswith(model):
            return PRICING[key]
    return _FALLBACK

File: pricing.go
package main

import "strings"

// Prices holds per-million-token rates for one model.
type Prices struct {
	Input     float64
	Cache5m   float64
	Cache1h   float64
	CacheRead float64
	Output    float64
}

// pricing maps model IDs to their rates.
// Source: https://platform.claude.com/docs/en/about-claude/pricing
var pricing = map[string]Prices{
	"claude-sonnet-4-6":         {3.0, 3.75, 6.0, 0.30, 15.0},
	"claude-sonnet-4-5":         {3.0, 3.75, 6.0, 0.30, 15.0},
	"claude-sonnet-4":           {3.0, 3.75, 6.0, 0.30, 15.0},
	"claude-opus-4-7":           {5.0, 6.25, 10.0, 0.50, 25.0},
	"claude-opus-4-6":           {5.0, 6.25, 10.0, 0.50, 25.0},
	"claude-opus-4-5":           {5.0, 6.25, 10.0, 0.50, 25.0},
	"claude-haiku-4-5":          {1.0, 1.25, 2.0, 0.10, 5.0},
	"claude-haiku-3-5-20241022": {0.80, 1.0, 1.6, 0.08, 4.0},
	"claude-haiku-3-5":          {0.80, 1.0, 1.6, 0.08, 4.0},
}

var fallbackPricing = Prices{3.0, 3.75, 6.0, 0.30, 15.0}

// getPricing returns the pricing for a model, falling back to Sonnet 4.6 rates.
func getPricing(model string) Prices {
	if p, ok := pricing[model]; ok {
		return p
	}
	for key, p := range pricing {
		if strings.HasPrefix(model, key) || strings.HasPrefix(key, model) {
			return p
		}
	}
	return fallbackPricing
}
